### ISSUE TITLE
Update request: squeak 5.3-19459 → 6.0-22148

### ISSUE THREAD

**Issue by suhr** [NixOS/nixpkgs#282414]
- Package name: squeak
- Latest released version: 6.0-22148

- Current version on the unstable channel: 5.3-19459
- Current version on the stable/release channel:

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**

@ehmry

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by ehmry** [comment]
Don't ping me, just update it yourself.

**Comment by suhr** [comment]
If you disable this patch, and fix other problems, you get the following error:

```
Q_SRC_VERSION=60 VM_MAJOR=6 VM_MINOR=0 VM_RELEASE=202312181441 VM_VERSION=6.0-202312181441
clean? checking for gcc... gcc
checking whether the C compiler works... no
configure: error: in '/build/source/building/linux64x64/squeak.cog.spur/build':
configure: error: C compiler cannot create executables
See 'config.log' for more details
```

**Comment by suhr** [comment]
@jbaum98 This is error on building opensmalltalk vm 202312181441. As the maintainer of the `opensmalltalk-vm` package you might consider to take a look.

`squeak` builds opensmalltalk vm itself, but it really should use `opensmalltalk-vm` package instead.

**Comment by ehmry** [comment]
Yes, we should merge what Squeak packages we have into one. I am listed as maintainer but I'm not using Squeak enough to make an update and test it thoroughly.

**Comment by jbaum98** [comment]
I tried updating opensmalltalk vm, and it needs a little work to build. I'm working on that.

After that's done, I'm open to trying to consolidate the squeak packages, but I also don't use squeak or smalltalk that much to feel confident messing around with what's already there on my own. That's why when I added the opensmalltalk vm packages I left everything else alone.
